forem/forem
# Allow ModRole & Admin to Suspend & Unsuspend User

## What type of PR is this? (check all applicable)


## Description
This PR brings the Suspend and Unsuspend User flows to the Actions Panel, allowing both admins and moderators to suspend or unsuspend a user from the Actions Panel.

## Related Tickets & Documents
- Closes https://github.com/forem/forem/issues/17446

## QA Instructions, Screenshots, Recordings
- As an admin, enable the moderator_role feature flag (/admin/feature_flags/features).
- Sign in as a moderator, navigate to a post and open the Actions Panel.
- Open the "Moderating actions" dropdown and click "Suspend <username>".
- In the Suspend modal, click the "Submit & Suspend" button. An error message should appear and the form should not submit.
- Now type a suspension reason then click "Submit & Suspend" button. The modal should close and you should get a snackbar confirmation that the user was suspended.
- As an admin, visit the suspended user's page in Member Manager. You should see the "suspended" flag next to their name. Check that you can still suspend users from the `/admin` side of things (as an admin).

### UI accessibility concerns?
Will be addressed as identified.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._
